src/key/factory.js

```diff
@@ -270,8 +270,8 @@ async function wrapKeyObject(secretKeyPacket, secretSubkeyPackets, options) {
  * @static
  */
 export async function read(data) {
-  const packetlist = new packet.List();
-  await packetlist.read(data);
+  const packetlist = new PacketList();
+  await packetlist.read(data, helper.allowedKeyPackets);
```

Comment:
do we need `allowedKeyPackets` here? Maybe instead of helper it is better to place them in Key class.

Reply:
Both `key.js` and `factory.js` need it, which is why I put it in `helper.js`.